src/main/java/com/github/cameltooling/lsp/internal/parser/CamelKModelineInsertionParser.java

```diff
@@ -0,0 +1,62 @@
+package com.github.cameltooling.lsp.internal.parser;
+
+import com.github.cameltooling.lsp.internal.completion.modeline.CamelKModelineInsertionProcessor;
+import org.eclipse.lsp4j.Position;
+import org.eclipse.lsp4j.TextDocumentItem;
+
+import java.io.File;
+import java.util.Arrays;
+import java.util.Optional;
+
+public class CamelKModelineInsertionParser {
+
+    public boolean canPutCamelKModeline(Position position, TextDocumentItem textDocumentItem) {
+        int currentLine = position.getLine();
+
+        //Save the lines to not have to access multiple times to file?
+        return isCamelKFile(textDocumentItem)
+                && lineIsEmpty(currentLine, textDocumentItem)
+                && noModelineInsertedAlready(textDocumentItem)
+                && previousLinesAreCommentsOrEmpty(currentLine,textDocumentItem);
+    }
+
+    private boolean isCamelKFile(TextDocumentItem document) {
+        return FileType.getFileTypeCorrespondingToUri(document.getUri()).isPresent();
+    }
+
+    private boolean lineIsEmpty(int line, TextDocumentItem textDocumentItem) {
+        return new ParserFileHelperUtil().getLine(textDocumentItem, line).isEmpty();
+    }
+
+    private boolean noModelineInsertedAlready(TextDocumentItem textDocumentItem) {
+        //Will do later
+        return true;
+    }
+
+    private boolean previousLinesAreCommentsOrEmpty(int line, TextDocumentItem textDocumentItem) {
+        //Harder than it looks. We have to take into account multi and single comment lines. Will be done when I get to that test
+        return true;
+    }
```

Comment:
i think that it is better to remove these methods if they are not implemented. Mentionign the scope in the readme/changelog/issues is enough. We never know when and if it will be finally implemented. Priorities can always change for one day to another.
Not writing it keep the code simpler.

Reply:
In this case they have reason to be because I absolutely need to fill those before shipping, and as a top to down programmer it lets me remember which use cases I have yet to define/fill.

I totally agree if I were to merge this to main and those functions were empty tho. No need in having placeholder functions that do nothing in prod code.

Reply:
I would like to always have the main branch to be releasable. For the Eclipse Desktop client, we are even in a Continuous delivery mode by picking the snapshots (so not very clean but continuous delivery anyway).
if you think that it is not releasable without it, we need to implement it in this Pull request before merging.

For this specific use case, i think that it is fine to merge and potentially release with completion appearing only for completely empty files. It won't cause any troubles for end users.

Reply:
Ok, don't worry because before the PR exits draft mode it will be filled :)

Btw I think I'll have finished by tomorrow. There is not much left.

I'll also update the gif with the different cases